### PR TITLE
Add windows support

### PR DESCRIPTION
**Pull request by StephenSmith25**
Adding basic windows support. I realise there's a PR up already, but its appears to have gone stale. I made a comment in the issue here #2, but I don't want to pester too much, as people are busy or have other more pressing things to do. This is likely less sophisticated than the other PR, but its working on my end. Its based off the anaconda recipe https://github.com/AnacondaRecipes/cyrus-sasl-feedstock.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by StephenSmith25** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -38,24 +44,32 @@ requirements:
...
+    - libntlm  # [unix]
+    - libdb  # [win]
     
 test:
   commands:
     - echo "DONE"
+    - if not exist %LIBRARY_LIB%\\libsasl.lib exit 1  # [win]
 
 about:
   home: https://www.cyrusimap.org/sasl/
```
Comment (wolfv) [inline recipe/meta.yaml:75]: maybe you could add yourself to the maintainer list?

Comment (StephenSmith25) [reply recipe/meta.yaml:75]: Thank you for taking a look at this @wolfv, I have added myself to the maintainer list, when you have a moment could you take another look?

Comment (StephenSmith25) [reply recipe/meta.yaml:75]: This slipped my mind, when you have a moment could you take another look @wolfv ?

**Pull request merged**